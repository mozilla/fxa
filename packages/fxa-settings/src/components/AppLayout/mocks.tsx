/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { RelierCmsInfo } from '../../models/integrations';

export const MOCK_CMS_INFO_VALID_LINEAR_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_VALID_RADIAL_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_INVALID_BG_COLOR = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'solid-color-red',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_WITH_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'CMS Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_NO_ALT = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_DEFAULT_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: undefined,
    logoAltText: undefined,
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_WITH_OTHER_PROPS = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/other-logo.png',
    logoAltText: 'Other Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const INTEGRATION_WITH_UNDEFINED_GETCMSINFO = {
  getCmsInfo: () => undefined,
};

export const INTEGRATION_WITH_NO_GETCMSINFO = {} as Record<string, never>;

export function createIntegration(cmsInfo?: RelierCmsInfo) {
  return {
    getCmsInfo: () => cmsInfo,
  };
}
