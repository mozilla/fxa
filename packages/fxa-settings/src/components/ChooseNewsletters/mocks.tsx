/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { useState } from 'react';
import ChooseNewsletters from '.';
import { newsletters } from './newsletters';

export const SubjectWithNewsletters = () => {
  const [, setSelected] = useState<string[]>([]);

  const newsletters = [
    {
      label: 'Early access to test new products',
      slug: ['test-pilot'],
      ftlId: 'choose-newsletters-option-test-pilot',
    },
  ];
  return (
    <ChooseNewsletters
      {...{ newsletters }}
      setSelectedNewsletterSlugs={setSelected}
    />
  );
};

export const SubjectWithNone = () => {
  const [, setSelected] = useState<string[]>([]);

  return (
    <ChooseNewsletters
      {...{ newsletters }}
      setSelectedNewsletterSlugs={setSelected}
    />
  );
};
