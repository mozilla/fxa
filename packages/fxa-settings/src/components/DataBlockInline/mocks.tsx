/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import DataBlockInline, { DataBlockInlineProps } from '.';

export const Subject = ({
  value = 'ANMD 1S09 7Y2Y 4EES 02CW BJ6Z PYKP H69F',
  ...props // overrides
}: Partial<DataBlockInlineProps> = {}) => (
  <DataBlockInline {...{ value, ...props }} />
);
