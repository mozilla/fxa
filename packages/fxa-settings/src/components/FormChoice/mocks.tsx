/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { CHOICES } from '.';
import {
  BackupAuthenticationCodesImage,
  BackupRecoveryPhoneSmsImage,
} from '../images';

export const commonFormChoiceProps = {
  isSubmitting: false,
  onSubmit: (data: any) => console.log('Submitted with choice:', data),
  legendEl: (
    <legend>
      <h2 className="card-header mb-4">I am (form) legend</h2>
    </legend>
  ),
  formChoices: [
    {
      id: 'recovery-choice-phone',
      value: CHOICES.phone,
      image: <BackupRecoveryPhoneSmsImage ariaHidden />,
      localizedChoiceTitle: '1st Choice Example',
      localizedChoiceInfo: '1st example description',
    },
    {
      id: 'recovery-choice-code',
      value: CHOICES.code,
      image: <BackupAuthenticationCodesImage ariaHidden />,
      localizedChoiceTitle: '2nd Choice Example',
      localizedChoiceInfo: '2nd example description',
    },
  ],
};

export const commonFormChoicePropsWithCms = {
  ...commonFormChoiceProps,
  cmsButton: {
    color: '#592ACB',
    text: 'Continue with CMS',
  },
};
