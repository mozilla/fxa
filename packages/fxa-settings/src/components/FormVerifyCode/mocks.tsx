/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import FormVerifyCode, { FormAttributes } from '.';
import { useFtlMsgResolver } from '../../models';

export const MOCK_EMAIL = 'test@example.com';

export const Subject = () => {
  type FormData = {
    code?: string;
  };

  const [code, setCode] = useState<string>('');
  const [codeErrorMessage, setCodeErrorMessage] = useState<string>('');
  const ftlMsgResolver = useFtlMsgResolver();

  const formAttributes: FormAttributes = {
    inputFtlId: 'demo-input-label-id',
    inputLabelText: 'Enter your 4-digit code',
    pattern: '[0-9]{4}',
    maxLength: 4,
    submitButtonFtlId: 'demo-submit-button-id',
    submitButtonText: 'Check that code',
  };

  const onFormSubmit = () => {
    if (!code) {
      const codeRequiredError = ftlMsgResolver.getMsg(
        'confirm-signup-code-required-error',
        'Confirmation code required'
      );
      setCodeErrorMessage(codeRequiredError);
    }
  };

  const { handleSubmit } = useForm<FormData>({
    mode: 'onBlur',
    criteriaMode: 'all',
    defaultValues: {
      code: '',
    },
  });

  return (
    <FormVerifyCode
      email={MOCK_EMAIL}
      onSubmit={handleSubmit(onFormSubmit)}
      viewName="default-view"
      {...{
        formAttributes,
        code,
        setCode,
        codeErrorMessage,
        setCodeErrorMessage,
      }}
    />
  );
};
