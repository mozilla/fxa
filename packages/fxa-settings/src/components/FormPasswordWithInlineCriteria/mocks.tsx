/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { useForm } from 'react-hook-form';
import FormPasswordWithInlineCriteria, { PasswordFormType } from '.';
import { MOCK_ACCOUNT } from '../../models/mocks';

type SubjectProps = {
  passwordFormType: PasswordFormType;
  requirePasswordConfirmation?: boolean;
};

export const Subject = ({
  passwordFormType,
  requirePasswordConfirmation,
}: SubjectProps) => {
  type FormData = {
    oldPassword?: string;
    newPassword: string;
    confirmPassword: string;
  };
  const onFormSubmit = () => {
    // this alert is for Storybook
    alert('Form submitted! (onFormSubmit called)');
  };

  const { handleSubmit, register, getValues, errors, formState, trigger } =
    useForm<FormData>({
      mode: 'onTouched',
      criteriaMode: 'all',
      defaultValues: {
        newPassword: '',
        confirmPassword: '',
      },
    });

  return (
    <FormPasswordWithInlineCriteria
      {...{
        formState,
        errors,
        trigger,
        register,
        getValues,
        passwordFormType,
        requirePasswordConfirmation,
      }}
      onSubmit={handleSubmit(onFormSubmit)}
      email={MOCK_ACCOUNT.primaryEmail.email}
      loading={false}
      onFocusMetricsEvent={() => {}}
    />
  );
};
