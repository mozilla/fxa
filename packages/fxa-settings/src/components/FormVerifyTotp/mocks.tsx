/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React, { useCallback, useState } from 'react';
import AppLayout from '../AppLayout';
import FormVerifyTotp from '.';
import { FormVerifyTotpProps } from './interfaces';

export const Subject = ({
  codeLength = 6,
  codeType = 'numeric',
  success = true,
  initialErrorMessage = '',
}: Partial<FormVerifyTotpProps> & {
  success?: Boolean;
  initialErrorMessage?: string;
}) => {
  const localizedInputLabel = `Enter ${codeLength.toString()}-digit code`;
  const localizedSubmitButtonText = 'Submit';
  const [errorMessage, setErrorMessage] = useState(initialErrorMessage);

  const mockVerifyCodeSuccess = useCallback(
    (code: string) => alert(`Mock code submission with code ${code}`),
    []
  );

  const mockVerifyCodeFail = useCallback(
    (code: string) => setErrorMessage('Something went wrong'),
    []
  );

  const verifyCode = success ? mockVerifyCodeSuccess : mockVerifyCodeFail;

  return (
    <AppLayout>
      <FormVerifyTotp
        {...{
          codeLength,
          codeType,
          errorMessage,
          localizedInputLabel,
          localizedSubmitButtonText,
          setErrorMessage,
          verifyCode,
        }}
      />
    </AppLayout>
  );
};

export default Subject;
