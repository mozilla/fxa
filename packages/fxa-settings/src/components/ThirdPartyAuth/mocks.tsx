/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { LocationProvider } from '@reach/router';
import ThirdPartyAuth, { ThirdPartyAuthProps } from '.';
import { AppContext } from '../../models';
import { mockAppContext } from '../../models/mocks';

export const Subject = (props: ThirdPartyAuthProps) => {
  return (
    <AppContext.Provider value={mockAppContext()}>
      <LocationProvider>
        <ThirdPartyAuth {...props} />
      </LocationProvider>
    </AppContext.Provider>
  );
};
