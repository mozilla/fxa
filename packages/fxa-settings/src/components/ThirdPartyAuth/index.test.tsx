/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
import { Subject } from './mocks';

const mockViewWithNoPasswordSet = jest.fn();
const mockStartGoogleAuthFromIndex = jest.fn();
const mockStartAppleAuthFromIndex = jest.fn();
const mockStartGoogleAuthFromLogin = jest.fn();
const mockStartAppleAuthFromLogin = jest.fn();
const mockStartGoogleAuthFromReg = jest.fn();
const mockStartAppleAuthFromReg = jest.fn();
const mockGleanIsDone = jest.fn();

jest.mock('../../lib/glean', () => {
  return {
    __esModule: true,
    default: {
      isDone: () => {
        mockGleanIsDone();
      },
      emailFirst: {
        googleOauthStart: () => {
          mockStartGoogleAuthFromIndex();
        },
        appleOauthStart: () => {
          mockStartAppleAuthFromIndex();
        },
      },
      thirdPartyAuth: {
        loginNoPwView: () => {
          mockViewWithNoPasswordSet();
        },
        startGoogleAuthFromLogin: () => {
          mockStartGoogleAuthFromLogin();
        },
        startAppleAuthFromLogin: () => {
          mockStartAppleAuthFromLogin();
        },
        startGoogleAuthFromReg: () => {
          mockStartGoogleAuthFromReg();
        },
        startAppleAuthFromReg: () => {
          mockStartAppleAuthFromReg();
        },
      },
    },
  };
});

function renderWith(props?: any) {
  return renderWithLocalizationProvider(<Subject {...props} />);
}

const mockFormSubmit = jest.fn();

describe('ThirdPartyAuthComponent', () => {
  // Form submission not supported in jest test. Instead, prevent the submission
  // and illustrate a 'short circuit' operation.
  const onSubmit = (e: any) => {
    e.preventDefault();
    return false;
  };
  const onContinueWithApple = jest.fn().mockImplementation(onSubmit);
  const onContinueWithGoogle = jest.fn().mockImplementation(onSubmit);

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders', async () => {
    renderWith({
      enabled: true,
    });

    await screen.findByText('Continue with Google');
    await screen.findByText('Continue with Apple');
    expect(mockFormSubmit).not.toBeCalled();

    expect(
      (await screen.findByTestId('google-signin-form-state')).getAttribute(
        'value'
      )
    ).toEqual('http%3A%2F%2Flocalhost%2F%3F');

    expect(
      (await screen.findByTestId('apple-signin-form-state')).getAttribute(
        'value'
      )
    ).toEqual('http%3A%2F%2Flocalhost%2F%3F');
  });

  it('submits apple form', async () => {
    renderWith({
      enabled: true,
      showSeparator: true,
      onContinueWithApple,
      onContinueWithGoogle,
    });

    const button = await screen.findByText('Continue with Apple');
    button.click();

    // Ensure the hidden input for state was updated.
    expect(
      (await screen.findByTestId('apple-signin-form-state')).getAttribute(
        'value'
      )
    ).not.toEqual('');
    expect(onContinueWithApple).toBeCalled();
    expect(onContinueWithGoogle).not.toBeCalled();
  });

  it('submits google form', async () => {
    renderWith({
      enabled: true,
      showSeparator: true,
      onContinueWithApple,
      onContinueWithGoogle,
    });

    const button = await screen.findByText('Continue with Google');
    button.click();

    // Ensure the hidden input for state was updated.
    expect(
      (await screen.findByTestId('google-signin-form-state')).getAttribute(
        'value'
      )
    ).not.toEqual('');
    expect(onContinueWithGoogle).toBeCalled();
    expect(onContinueWithApple).not.toBeCalled();
  });

  it('should deeplink directly to google auth, if deeplink=`googleLogin`', async () => {
    renderWith({
      enabled: true,
      showSeparator: false,
      deeplink: 'googleLogin',
      view: 'index'
    });

    expect(
      (await screen.findByTestId('google-signin-form-state')).getAttribute(
        'value'
      )
    ).not.toEqual('');
  })

  it('should deeplink directly to apple auth, if deeplink=`appleLogin`', async () => {
    renderWith({
      enabled: true,
      showSeparator: false,
      deeplink: 'appleLogin',
      view: 'index'
    });

    expect(
      (await screen.findByTestId('apple-signin-form-state')).getAttribute(
        'value'
      )
    ).not.toEqual('');
  })

  it('hides separator', async () => {
    renderWith({
      enabled: true,
      showSeparator: false,
      onContinueWithApple,
      onContinueWithGoogle,
    });

    expect(screen.queryByText('Or')).toBeNull();
    expect(mockViewWithNoPasswordSet).toBeCalled();
  });

  it('shows separator', async () => {
    renderWith({
      enabled: true,
      showSeparator: true,
      onContinueWithApple,
      onContinueWithGoogle,
    });

    expect(screen.queryByText('Or')).toBeDefined();
    expect(mockViewWithNoPasswordSet).not.toBeCalled();
  });

  describe('emits metrics', () => {
    it('emits glean metrics loginNoPwView', () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
      });
      expect(mockViewWithNoPasswordSet).toBeCalled();
    });

    it('emits glean metrics startGoogleAuthFromIndex', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'index',
      });
      const button = await screen.findByText('Continue with Google');
      button.click();
      expect(mockStartGoogleAuthFromIndex).toBeCalled();
      expect(mockGleanIsDone).toBeCalled();
    });

    it('emits glean metrics startAppleAuthFromIndex', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'index',
      });
      const button = await screen.findByText('Continue with Apple');
      button.click();
      expect(mockStartAppleAuthFromIndex).toBeCalled();
    });

    it('emits glean metrics startGoogleAuthFromLogin', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'signin',
      });
      const button = await screen.findByText('Continue with Google');
      button.click();
      expect(mockStartGoogleAuthFromLogin).toBeCalled();
      expect(mockGleanIsDone).toBeCalled();
    });

    it('emits glean metrics startAppleAuthFromLogin', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'signin',
      });
      const button = await screen.findByText('Continue with Apple');
      button.click();
      expect(mockStartAppleAuthFromLogin).toBeCalled();
    });

    it('emits glean metrics startGoogleAuthFromReg', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'signup',
      });
      const button = await screen.findByText('Continue with Google');
      button.click();
      expect(mockStartGoogleAuthFromReg).toBeCalled();
      expect(mockGleanIsDone).toBeCalled();
    });

    it('emits glean metrics startAppleAuthFromReg', async () => {
      renderWith({
        enabled: true,
        showSeparator: false,
        onContinueWithApple,
        onContinueWithGoogle,
        viewName: 'signup',
      });
      const button = await screen.findByText('Continue with Apple');
      button.click();
      expect(mockStartAppleAuthFromReg).toBeCalled();
      expect(mockGleanIsDone).toBeCalled();
    });
  });
});
