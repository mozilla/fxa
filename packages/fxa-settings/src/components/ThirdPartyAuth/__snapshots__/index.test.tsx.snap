// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdPartyAuthComponent buttons match snapshot: apple 1`] = `
<button
  aria-label="Continue with Apple"
  class="w-[60px] h-[60px] p-4 flex items-center justify-center rounded-full border focus-visible-default outline-offset-2
          bg-black border-transparent
          "
  type="submit"
>
  <svg>
    apple-logo-viewbox-white.svg
  </svg>
</button>
`;

exports[`ThirdPartyAuthComponent buttons match snapshot: google 1`] = `
<button
  aria-label="Continue with Google"
  class="w-[60px] h-[60px] p-4 flex items-center justify-center rounded-full border focus-visible-default outline-offset-2
          bg-[#F9F4F4] border-[#747775] border-[1px]
          "
  type="submit"
>
  <svg>
    google-logo-viewbox.svg
  </svg>
</button>
`;
